src/main/java/frc/robot/subsystems/RollerSubsystem.java

```diff
@@ -145,15 +160,17 @@ public void periodic() {
 
         HeldPiece limitPiece = getLimitSwitchPiece();
         HeldPiece proximityPiece = getProximitySensorPiece();
+        HeldPiece colorPiece = getColorSensorPiece();
 
         // If either the limit switch or the proximity sensor have detected a piece, set
         // the piece to the detected piece, prioritizing the proximity sensor over the limit
         // switch.
-        if (proximityPiece != HeldPiece.EMPTY) {
-            heldPiece = proximityPiece;
+        if (colorPiece != HeldPiece.EMPTY) {
```

Comment:
Do we want to do a 3-layer fallback here where we primarily use the color piece, fall back on the proximity piece, and finally the limit piece?

Reply:
alternatively, we shoudl remove call getProximityPiece.

layer of redundancy could be good but i'm also not sure if that'll take double the (already-long) time to poll the color sensor